packages/react-provider/src/graph-ql/index.ts

```diff
@@ -1,9 +1,9 @@
-import { useMemo } from "react";
-import { createClient } from "urql";
+import { Client, createClient } from "urql";
 
-export default ({ clientKey, userId, userSignature, apiUrl }) => {
-  const client = useMemo(() => {
-    return createClient({
+export class GraphQLClient {
+  client: Client;
+  constructor({ clientKey, userId, userSignature, apiUrl }) {
```

Comment:
what happens if the userId changes?

Reply:
If any one these variables change a new client will be created. Check here https://github.com/trycourier/courier-react/blob/d66fbf0e9f59eb5fdc7b8341487fa20729a50863/packages/react-provider/src/index.tsx#L36